mcstatus/__main__.py

```diff
@@ -1,75 +1,104 @@
 from __future__ import annotations
 
+import sys
 import argparse
 import socket
+from typing import TYPE_CHECKING
 from json import dumps as json_dumps
 
-from mcstatus import JavaServer
+from mcstatus import JavaServer, BedrockServer
+from mcstatus.responses import JavaStatusResponse
 
+if TYPE_CHECKING:
+    SupportedServers = JavaServer | BedrockServer
 
-def ping(server: JavaServer) -> None:
-    print(f"{server.ping()}ms")
 
+def ping(server: SupportedServers) -> int:
+    print(f"{server.status().latency}")  # XXX: JavaServer.ping() not working?
+    return 0
 
-def status(server: JavaServer) -> None:
+
+def status(server: SupportedServers) -> int:
     response = server.status()
-    if response.players.sample is not None:
-        player_sample = str([f"{player.name} ({player.id})" for player in response.players.sample])
+
+    java_res = response if isinstance(response, JavaStatusResponse) else None
+
+    if not java_res:
+        player_sample = ""
+    elif java_res.players.sample is not None:
+        player_sample = str([f"{player.name} ({player.id})" for player in java_res.players.sample])
     else:
         player_sample = "No players online"
 
-    print(f"version: v{response.version.name} (protocol {response.version.protocol})")
-    print(f'motd: "{response.motd}"')
-    print(f"players: {response.players.online}/{response.players.max} {player_sample}")
+    if player_sample:
+        player_sample = " " + player_sample
 
+    print(f"version: {server.kind()} {response.version.name} (protocol {response.version.protocol})")
+    print(f"motd: {response.motd.to_ansi()}")
+    print(f"players: {response.players.online}/{response.players.max}{player_sample}")
+    print(f"ping: {response.latency:.2f} ms")
+    return 0
 
-def json(server: JavaServer) -> None:
+
+def json(server: SupportedServers) -> int:
     data = {}
     data["online"] = False
+    data["kind"] = server.kind()
     # Build data with responses and quit on exception
     try:
         status_res = server.status(tries=1)
+        java_res = status_res if isinstance(status_res, JavaStatusResponse) else None
         data["version"] = status_res.version.name
         data["protocol"] = status_res.version.protocol
         data["motd"] = status_res.motd.raw
```

Comment:
Can you change it to this please?
```suggestion
        data["motd"] = status_res.motd.minecraft
```

Reply:
Is there much use in deserialising and reserialising the MOTD to the same format it already was? I don't feel strongly, though.

Reply:
Most of responses return motd as dict